proto/clustermetadata.proto

```diff
@@ -0,0 +1,205 @@
+  repeated string cells = 4;              
+}
+
+// =============================================================================
+// Cell Topology
+// =============================================================================
+
+// MultiPooler represents metadata about a running multipooler component instance in the cluster.
+message MultiPooler {
+  // identifier is the unique identifier of the multipooler in the cluster.
+  ID identifier = 1;
+
+  // Database name.
+  string database = 2;
+  
+  // TableGroup name.
+  string table_group = 3;
+
+  // Shard name. If range based sharding is used, it should match
+  // key_range.
+  string shard = 4;
+  
+  // If range based sharding is used, range for the pooler's shard.
+  KeyRange key_range = 5;
+
+  // PoolerType is the current type of the pooler.
+  PoolerType type = 6;
+  
+  // PoolerServingStatus is the current type of the pooler.
+  PoolerServingStatus serving_status = 7;
+
+  // Fully qualified domain name of the host.
+  string hostname = 8;
+  
+  // Map of named ports. Normally this should include postgres and grpc.
+  map<string, int32> port_map = 9;
+  
+  // tags
+  map<string, string> tags = 10;
+}
+
+// MultiGateway represents metadata about a running multigateway component instance in the cluster.
+message MultiGateway {
+  // identifier is the unique name of the multi gateway in the cluster.
+  ID identifier = 1;
+
+  // Fully qualified domain name of the host.
+  string hostname = 2;
+  
+  // Map of named ports. Normally this should include postgres and grpc.
+  map<string, int32> port_map = 3;
+}
+
+// MultiOrc represents information about a running instance of multiorc.
+message MultiOrc {
+   // identifier is the unique name of the MultiOrc in the cluster.
+  ID identifier = 1;
+  
+}
+
+// =============================================================================
+// Auxiliary Data Types
+// =============================================================================
+
+// PoolerID is a globally unique pooler identifier.
+message ID {
+  // component that this ID is used for (i.e multigateway / multiorc / multipooler)
+  MultigresComponent component = 1;
+  
+  // cell is the cell the pooler is located
+  string cell = 2;
+
+  // uid is a unique id for this pooler within the shard
+  // (this is the Postgres server id as well).
+  uint32 uid = 3;
+}
+
+// PoolerType represents the type of a given MultiPooler.
+enum PoolerType {
+  // UNKNOWN this is not a valid type. 
+  // It is useful during upgrades to detect when a value is not initialized
+  UNKNOWN = 0;
+
+  // PRIMARY is the primary server for the shard. Only PRIMARY allows DMLs.
+  PRIMARY = 1;
+
+  // REPLICA replicates from leader. It is used to read only traffic
+  REPLICA = 2;
+}
+
+// PoolerServingStatus represents the serving status of the given MultiPooler.
+enum PoolerServingStatus {
+  // SERVING is the status a server during normal operations when is serving traffic.
+  SERVING = 0;
```

Comment:
Probably all enums should have UNKNOWN as `PoolerType` does. See https://protobuf.dev/best-practices/dos-donts/#unspecified-enum